custom_components/srgssr_weather/config_flow.py

```diff
@@ -51,26 +52,24 @@ async def async_step_credentials(self, user_input: dict = None) -> dict:
             errors=errors,
         )
 
-    async def async_step_user(self, user_input: dict = None) -> dict:
-        if self._credentials is None:
-            return await self.async_step_credentials()
-        else:
-            return await self.async_step_location()
-
     async def async_step_location(self, user_input: dict = None) -> dict:
-        errors = {}
+        errors: dict[str, str] = {}
```

Comment:
Correct me if I'm wrong but I think Home Assistant only required Python 3.8 and PEP 585 is part of 3.9.
You either need to use `Dict[str, str]` and `from typing import Dict` or keep it as is and `from __future__ import annotations`